crates/bevy_utils/src/parallel_queue.rs

```diff
@@ -9,6 +13,37 @@ pub struct Parallel<T: Send> {
     locals: ThreadLocal<Cell<T>>,
 }
 
+/// A scope guard of a `Parallel`, when this struct is dropped ,the value will writeback to its `Parallel`
+pub struct ParallelGuard<'a, T: Send + Default> {
```

Comment:
I am also not a fan of using this drop implementation because it provides a gun to the user. In theory, another method to achieve the same performance is by using `par_chunk`. It is much safer and clearer, but it requires some refactoring in the `QueryParIter `.